File: CwRetail.Ui/src/models/product.ts
import { ProductType } from "../enumerations/product-type";

export class Product {
  constructor(public productId: bigint = 0n, public name: string = '', public price: number = 0, public type: ProductType = ProductType.Books, public active: boolean = false, public lastUpdated: string = '') { }
}

File: CwRetail.Data/Models/Product.cs
using CwRetail.Data.Enumerations;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace CwRetail.Data.Models
{
    public class Product
    {
        public long ProductId { get; set; }

        //max 100 characters
        public string Name { get; set; }

        //2 decimals
        public decimal Price { get; set; }

        public ProductTypeEnum Type { get; set; }

        public bool Active { get; set; }
    }
}
